### ISSUE TITLE
set7

### ISSUE THREAD

**Issue by MrinmoyBANIKYA** [MrinmoyBANIKYA/Assignment-#2]
# set 7 solutions
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.preprocessing import LabelEncoder
from sklearn.metrics import accuracy_score

df = pd.read_excel("SET-7.xlsx")

print("Dataset Preview:\n", df.head())

print("\nMissing Values:\n", df.isnull().sum())

X = df[['ground_clearance', 'height']]
y = df['category']

label_encoder = LabelEncoder()
y_encoded = label_encoder.fit_transform(y)

X_train_70, X_test_30, y_train_70, y_test_30 = train_test_split(X, y_encoded, test_size=0.3, random_state=42)
model_70 = LogisticRegression()
model_70.fit(X_train_70, y_train_70)
y_pred_30 = model_70.predict(X_test_30)
accuracy_70_30 = accuracy_score(y_test_30, y_pred_30)

X_train_80, X_test_20, y_train_80, y_test_20 = train_test_split(X, y_encoded, test_size=0.2, random_state=42)
model_80 = LogisticRegression()
model_80.fit(X_train_80, y_train_80)
y_pred_20 = model_80.predict(X_test_20)
accuracy_80_20 = accuracy_score(y_test_20, y_pred_20)

print(f"\n Accuracy with 70%-30% split: {accuracy_70_30:.2f}")
print(f" Accuracy with 80%-20% split: {accuracy_80_20:.2f}")
